### ISSUE TITLE
Question: Can you add descriptions of the various configuration options?

### ISSUE THREAD

**Issue by mDalessandro** [istanbuljs/nyc#904]
Hello! I'm trying to configure my .nycrc file, and the README doesn't really explain the various config properties.

Is there any way you can expand the README, or perhaps add a tutorial on https://istanbul.js.org/docs/tutorials/, to explain all the available configuration options and what they do? 

One of my immediate questions is on the "extension" property. Does adding .jsx there mean I don't have to add a second line to my "include" property?

In short, does this:
`"extension": [
      ".jsx"
    ],`

make the .jsx inclusion unecessary?
`"include": [
      "src/**/*.js",
      "src/**/*.jsx"
    ],`

**Comment by JaKXz** [comment]
Hi @mDalessandro sorry for the late reply. You can try `npx nyc --help` to see a verbose description of the options, hopefully that's what you're looking for.

To answer your question, I think in this case you would actually lose `src/**/*.js` since the `extension` option overrides the default `[".js"]`. The typical way to do this is:

```json
{
  "extension": [
    ".js",
    ".jsx"
  ],
  "include": [
    "src"
  ]
}
```

as test files are excluded by default.

I'm going to close this issue for now; if you have more specific questions, feel free to open new issues at: https://github.com/istanbuljs/istanbuljs.github.io and I can help you write content for the tutorials or a new section on all the options or something. I also think most of it is documented on the README, but you are also welcome to make PRs to that with the information you think would be helpful.
